src/Utility/Log.cpp

```diff
@@ -1,59 +1,148 @@
+            throw ("Error: i " + std::to_string(value));
+    #endif
+
     return *this;
 }
 
-Log& Log::operator <<(unsigned int value) {
-    fprintf(stderr, "%u", value);
+Log& Log::operator <<(const unsigned int value) {
+    *streams[currentChannel] << value;
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: u " + std::to_string(value));
+    #endif
+
     return *this;
 }
 
-Log& Log::operator <<(float value) {
-    fprintf(stderr, "%f", value);
+Log& Log::operator <<(const float value) {
+    *streams[currentChannel] << value;
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: f " + std::to_string(value));
+    #endif
+
     return *this;
 }
 
-Log& Log::operator <<(double value) {
-    fprintf(stderr, "%f", value);
+Log& Log::operator <<(const double value) {
+    *streams[currentChannel] << value;
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: d " + std::to_string(value));
+    #endif
+
     return *this;
 }
 
-Log& Log::operator <<(time_t value) {
+Log& Log::operator <<(const time_t value) {
     struct tm * timeinfo = localtime(&value);
-    
-    char buffer[24];
-    strftime(buffer, 24, "%Y-%m-%d %H:%M:%S", timeinfo);
-    *this << buffer;
-    
+
+    const unsigned int bufferLength = 24;
+
+    char buffer[bufferLength] = {'\0'};
+    strftime(buffer, bufferLength, "%Y-%m-%d %H:%M:%S", timeinfo);
+    string const outString = string(buffer, bufferLength);
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: t " + outString);
+    #endif
+
+    *streams[currentChannel] << outString;
+
     return *this;
 }
 
 Log& Log::operator <<(const glm::vec2& value) {
-    fprintf(stderr, "(%f, %f)", value.x, value.y);
+    *streams[currentChannel] << "(" << value.x << "," << value.y << ")";
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: vec2 x: " + std::to_string(value.x) + " y: " + std::to_string(value.y));
+    #endif
+
     return *this;
 }
 
 Log& Log::operator <<(const glm::vec3& value) {
-    fprintf(stderr, "(%f, %f, %f)", value.x, value.y, value.z);
+    *streams[currentChannel] << "(" << value.x << "," << value.y << "," << value.z << ")";
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: vec3 x: " + std::to_string(value.x) + " y: " + std::to_string(value.y) + " z: " + std::to_string(value.z));
+    #endif
+
     return *this;
 }
 
 Log & Log::operator<<(const glm::vec4 & value) {
-    fprintf(stderr, "(%f, %f, %f, %f)", value.x, value.y, value.z, value.w);
+    *streams[currentChannel] << "(" << value.x << "," << value.y << "," << value.z << "," << value.w << ")";
+
+    #ifdef LOGTESTING 
+        if (currentChannel == ERR)
+            throw ("Error: vec3 x: " + std::to_string(value.x) + " y: " + std::to_string(value.y) + " z: " + std::to_string(value.z) + " w: " + std::to_string(value.w));
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'throw' not immediately followed by a semicolon or a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT006)